src/components/main-view/main-view.jsx

```diff
@@ -16,107 +20,138 @@ export class MainView extends React.Component {
+    if (accessToken !== null) {
+      this.setState({
+        user: localStorage.getItem("user"),
+      });
+      this.getMovies(accessToken);
+    }
   }
 
-  componentDidMount() {
+
+  getMovies(token) {
     axios
-      .get("https://movie-api-1684.herokuapp.com/movies")
+      .get("https://movie-api-1684.herokuapp.com/movies", {
+        headers: { Authorization: `Bearer ${token}` },
+      })
       .then((response) => {
-        // Assign the result to the state
-        this.setState({ movies: response.data });
+        this.setState({
+          movies: response.data,
+        });
       })
       .catch(function (error) {
         console.log(error);
       });
   }
 
-  onMovieClick(movie) {
-    this.setState({ selectedMovie: movie });
-  }
-  goBack() {
-    this.setState({ selectedMovie: null });
-  }
-  onLoggedIn(user) {
-    this.setState({ user });
-  }
-  startRegister() {
-    this.setState({ register : true });
-  }
-  endRegister() {
-    this.setState({ register : null });
+  
+  // props.onLoggedIn(data) - LoginView
+  onLoggedIn(authData) {
+    console.log(authData);
+    this.setState({
+      user: authData.user.Username,
+    });
+
+    localStorage.setItem("token", authData.token);
+    localStorage.setItem("user", authData.user.Username);
+    this.getMovies(authData.token);
   }
 
-  render() {
-    const { movies, selectedMovie, user, register } = this.state;
+  onLoggedOut() {
+		localStorage.removeItem('token');
+		localStorage.removeItem('user');
+		this.setState({
+			user: null,
+		});
+	}
 
-   if (register) {
-      return <RegistrationView endRegister={() => this.endRegister()} />;
-    } 
-
-    if (!user)
-      return  <LoginView startRegister={() => this.startRegister()}
-      onLoggedIn={(user) => this.onLoggedIn(user)}/>
+  render() {
+    const { movies, user } = this.state;
 
     // Before the movies have been loaded
     if (!movies) return <div className="main-view" />;
 
     return (
-      <div className="main-view">
-      <Container>
-      <Row className="justify-content-md-center">
-        {selectedMovie ? (          
-            <Col md={8}>
-          <MovieView movie={selectedMovie} goBack={() => this.goBack()} />
-          </Col>
-        
-        ) : (
-          movies.map((movie) => (
-            <MovieCard
-              key={movie._id}
-              movie={movie}
-              onClick={(movie) => this.onMovieClick(movie)}
-            />
-          ))
-        )}
-        </Row>
-      </Container>
+      <Router>
+        <div className="main-view">
+          <h2>Main View</h2>
+
+        <ul>
+          <li>
+          <Route path="/register" render={() => <RegistrationView />} />
+          <Link to="/register">Register</Link>
+          </li>
+
+        <li>
+          <Route exact path="/movies/:movieId" render={({match}) => 
+          <MovieView movie={movies.find(m => m._id === match.params.movieId)}/>}/>
+          <Link to="/movies/:movieId">Movies</Link>
```

Comment:
`<Link to="/movies/:movieId">Movies</Link>`:
This does not work this way. React needs to know here what the parameter `:movieId` is (what movie do you want to render data from). Only register the route, just remove the <Link> part.

Reply:
Ok I've removed <Link....>Movies</Link>

Reply:
Ok I've removed the `<Link to="/movies/:movieId">Movies</Link>`